src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,62 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private K[] keyList;
+    private V[] valueList;
```

Comment:
* initialize these fields in the constructor and remove all of the redundant arrays null checks after this change
*  add a `size` variable which will represent the number of elements that we have put into our storage \+ a default max capacity constant
* this is not a `List` so let's not name it like it is
```suggestion
    private K[] keys;
    private V[] values;
```